src/redux/reducers/index.js

```diff
@@ -166,6 +148,19 @@ export const collectionsReducer = (state = preloadCollections, action) => {
         draft[collectionId].resourceTypeFilters[resourceType] = !prevValue;
       });
     }
+    case actionTypes.UPDATE_DATE_RANGE_FILTER: {
+      const { collectionId, dateRangeStart, dateRangeEnd } = action.payload;
+      return produce(state, (draft) => {
+        if (dateRangeStart) {
+          // eslint-disable-next-line no-param-reassign
```

Comment:
should we just whitelist this rule? seems like we'll be using it a lot

Reply:
it helps prevent unintended mutation in variety of places; only these specific cases are exceptions b/c of `immer`   (also: interval iterator in `timelineIntervalsSelector` )